src/patterns.md

```diff
@@ -0,0 +1,646 @@
+    panic!("Shouldn't reach here");
+} else if let (a, 4) = (3, 4) {    // "(a, 4)" is refutable, and will match
+    println!("Matched ({}, 4)", a);
+}
+```
+
+## Literal patterns
+
+> **<sup>Syntax</sup>**\
+> _LiteralPattern_ :\
+> &nbsp;&nbsp; &nbsp;&nbsp; [BOOLEAN_LITERAL]\
+> &nbsp;&nbsp; | [CHAR_LITERAL]\
+> &nbsp;&nbsp; | [BYTE_LITERAL]\
+> &nbsp;&nbsp; | [STRING_LITERAL]\
+> &nbsp;&nbsp; | [RAW_STRING_LITERAL]\
+> &nbsp;&nbsp; | [BYTE_STRING_LITERAL]\
+> &nbsp;&nbsp; | [RAW_BYTE_STRING_LITERAL]\
+> &nbsp;&nbsp; | `-`<sup>?</sup> [INTEGER_LITERAL]\
+> &nbsp;&nbsp; | `-`<sup>?</sup> [FLOAT_LITERAL]
+
+[BOOLEAN_LITERAL]: tokens.html#boolean-literals
+[CHAR_LITERAL]: tokens.html#character-literals
+[BYTE_LITERAL]: tokens.html#byte-literals
+[STRING_LITERAL]: tokens.html#string-literals
+[RAW_STRING_LITERAL]: tokens.html#raw-string-literals
+[BYTE_STRING_LITERAL]: tokens.html#byte-string-literals
+[RAW_BYTE_STRING_LITERAL]: tokens.html#raw-byte-string-literals
+[INTEGER_LITERAL]: tokens.html#integer-literals
+[FLOAT_LITERAL]: tokens.html#floating-point-literals
+
+_Literal patterns_ match exactly the value they represent. Since negative numbers are
+not literals in Rust, literal patterns also accept an optional minus sign before the
+literal.
+
+Floating-point literals are currently accepted, but due to the complexity of comparing
+them, they are going to be forbidden on literal patterns in a future version of Rust (see
+[issue #41620](https://github.com/rust-lang/rust/issues/41620)).
+
+Literal patterns are always refutable.
+
+Examples:
+
+```rust
+for i in -2..5 {
+    match i {
+        -1 => println!("It's minus one"),
+        1 => println!("It's a one"),
+        2|4 => println!("It's either a two or a four"),
+        _ => println!("Matched none of the arms"),
+    }
+}
+```
+
+## Identifier patterns
+
+> **<sup>Syntax</sup>**\
+> _IdentifierPattern_ :\
+> &nbsp;&nbsp; &nbsp;&nbsp; `ref`<sup>?</sup> `mut`<sup>?</sup> [IDENTIFIER] (`@` [_Pattern_] ) <sup>?</sup>
+
+_Identifier patterns_ bind the value they match to a **previously undeclared** variable.
+
+Patterns that consist of only an identifier, possibly with a `mut`, like
+`variable`, `x`, and `y` below:
+
+```rust
+let mut variable = 10;
+fn sum(x: i32, y: i32) -> i32 {
+#    x + y
+# }
+```
+
+match any value and bind it to that identifier. This is the most commonly
+used pattern in variable declarations and function/closure parameters.
+
+To bind non-trivial patterns to a variable, the use of the syntax `variable @
+subpattern` is needed. For example:
+
+```rust
+let x = 2;
+
+match x {
+    e @ 1 ..= 5 => println!("got a range element {}", e),
+    _ => println!("anything"),
+}
+```
+
+binds to `e` the value 2 (not the entire range: the range here is a range subpattern).
+
+By default, identifier patterns bind a variable to a copy of or move from the
+matched value (depending whether the matched value implements the [Copy trait]).
```

Comment:
* The parenthesis are superfluous. The words inside are crucial to the meaning of the sentence. 
* "depending" -> "depending on".
* `the [Copy trait]` -> ``[`Copy`]``